### ISSUE TITLE
configure: WARNING: unrecognized options: --with-compiler

### ISSUE THREAD

**Issue by lyokha** [haskell-hvr/resolv#20]
Cabal unconditionally passes option `--with-compiler` or `--with-hc` in `Distribution.Simple.Setup.configureArgs`. This makes `configure` run from `cabal v1-configure` or `cabal v2-build` print the warning.

This can be fixed by declaring dummy *with-options* in configure.ac

```diff
diff --git a/configure.ac b/configure.ac
index 93942e2..1a1c036 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7,6 +7,10 @@ AC_CONFIG_HEADERS([cbits/hs_resolv_config.h])
 AC_PROG_CC
 AC_C_CONST
 
+dnl ignore options --with-compiler and --with-hc passed by cabal configure
+AC_ARG_WITH([compiler], [], [], [])
+AC_ARG_WITH([hc], [], [], [])
+
 AC_CHECK_HEADERS([netinet/in.h arpa/nameser.h])
 AC_CHECK_HEADER([resolv.h],,AC_MSG_ERROR([required <resolv.h> header not found]),[[
 #include <sys/types.h>
```

**Comment by andreasabel** [comment]
Unfortunately, I do not know anything about the `Configure` mode of `cabal`.  I found some upstream issues:
- https://github.com/haskell/cabal/issues/2947
- https://github.com/haskell/cabal/issues/2801
- https://github.com/haskell/cabal/issues/7452

**Comment by lyokha** [comment]
Thanks for the links! I found a good comment reading through them: https://github.com/haskell/cabal/pull/2946#issuecomment-164294686. It says that a few packages want to "shut up" this warning by supplying macro definitions similar to what I proposed (except that I skip the help message). This all makes sense so far as the configure script makes no adjustments related to haskell compiler (as in the resolv's configure, and this is why it complains about the option) and only tests and configures C bits.
